configs/components/ruby-2.7.8.rb

```diff
@@ -106,7 +106,10 @@
     special_flags += " --build=#{settings[:platform_triple]} "
   elsif platform.is_cross_compiled? && platform.is_linux?
     special_flags += " --with-baseruby=#{host_ruby} "
-  elsif platform.is_cross_compiled? && platform.is_macos?
+  elsif platform.is_macos? && platform.architecture == 'arm64'
+    if platform.os_version.to_i >= 13
+      pkg.environment 'PATH', '/opt/homebrew/bin:$(PATH):/usr/local/bin'
```

Comment:
Suggest adding this for macOS 13+ ARM only for now:

```ruby      
      special_flags += " --with-openssl-dir=#{settings[:prefix]} "
```

And then file a ticket to investigate adding to all native builds later:

```ruby
special_flags = " --prefix=#{ruby_dir} --with-opt-dir=#{settings[:prefix]} "
special_flags += "--with-openssl-dir=#{settings[:prefix]} " unless platform.is_cross_compiled?
```